File: src/main/java/cn/xunyard/idea/coding/i18n/logic/InspectionUtils.java
package cn.xunyard.idea.coding.i18n.logic;

import com.intellij.psi.PsiElement;
import com.intellij.psi.PsiLiteralExpression;

/**
 * @author <a herf="mailto:wuqi@terminus.io">xunyard</a>
 * @date 2020-01-11
 */
public class InspectionUtils {

    public static boolean isErrorCode(String str) {
        if (!str.startsWith("\"") || !str.endsWith("\"")) {
            return false;
        }

        boolean containsDot = false;
        char[] chars = str.toCharArray();
        for (int i = 1; i < chars.length - 2; i++) {
            char c = chars[i];

            if (c == '.') {
                containsDot = true;
                continue;
            }

            if (c > 'a' - 1 && c < 'z' + 1) {
                continue;
            }

            if (c > 'A' - 1 && c < 'Z' + 1) {
                continue;
            }

            return false;
        }

        return containsDot;
    }

    public static String getErrorCode(PsiElement psiElement) {
        if (!(psiElement instanceof PsiLiteralExpression)) {
            return null;
        }

        String text = psiElement.getText();
        if (text.length() < 3) {
            return null;
        }

        return text.substring(1, text.length() - 1);
    }

}

File: src/main/kotlin/cn/xunyard/idea/coding/i18n/logic/InspectionUtils.kt
package cn.xunyard.idea.coding.i18n.logic

import com.intellij.psi.PsiElement
import com.intellij.psi.PsiLiteralExpression

/**
 * @author <a herf="mailto:wuqi@terminus.io">xunyard</a>
 * @date 2020-01-11
 */
object InspectionUtils {
    fun isErrorCode(str: String): Boolean {
        if (!str.startsWith("\"") || !str.endsWith("\"")) {
            return false
        }
        var containsDot = false
        val chars = str.toCharArray()
        for (i in 1 until chars.size - 2) {
            val c = chars[i]
            if (c == '.') {
                containsDot = true
                continue
            }
            if (c.toInt() > 'a'.toInt() - 1 && c.toInt() < 'z'.toInt() + 1) {
                continue
            }
            if (c.toInt() > 'A'.toInt() - 1 && c.toInt() < 'Z'.toInt() + 1) {
                continue
            }
            return false
        }
        return containsDot
    }

    fun getErrorCode(psiElement: PsiElement): String? {
        if (psiElement !is PsiLiteralExpression) {
            return null
        }
        val text = psiElement.getText()
        return if (text.length < 3) {
            null
        } else text.substring(1, text.length - 1)
    }
}